src/ucp/tag/eager_snd.c

```diff
@@ -36,11 +68,13 @@ static size_t ucp_tag_pack_eager_sync_only_contig(void *dest, void *arg)
     hdr->req.sender_uuid = req->send.ep->worker->uuid;
     hdr->req.reqptr      = (uintptr_t)req;
 
-    memcpy(hdr + 1, req->send.buffer, length);
+    ucs_assert(req->send.state.offset == 0);
+    ucp_tag_pack_eager_dt_copy(hdr + 1, req->send.buffer, &req->send.state,
```

Comment:
i would expect only the buffer itself (which is pointed by iov) could not be reused. which is the case in UCT. here we also say the IOV array which specifies the buffers cannot be reused as well.